realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -286,6 +286,9 @@ private void emitAccessors(JavaWriter writer) throws IOException {
                     writer.emitStatement("row.nullifyLink(%s)", fieldIndexVariableReference(field));
                     writer.emitStatement("return");
                 writer.endControlFlow();
+                writer.beginControlFlow("if (!value.isValid())");
```

Comment:
You also need to check if the object is in the same Realm. This would also behaves really strange if you tried to add an object from another Realm.